geomesa-dynamo/geomesa-dynamodb-datastore/src/main/scala/org/locationtech/geomesa/dynamodb/data/DynamoDBDataStore.scala

```diff
@@ -0,0 +1,239 @@
+    val tableName = makeSFTTableName(catalog, nameSFT)
+    val table = dynamoDB.getTable(tableName)
+    val currentPT = table.getDescription.getProvisionedThroughput
+    val r: Long = rcus.getOrElse(currentPT.getReadCapacityUnits)
+    val w: Long = wcus.getOrElse(currentPT.getWriteCapacityUnits)
+    setPTforSFT(table, r, w)
+  }
+
+  def setReadsSFT(nameSFT: String, rcus: Long): Unit = {
+    applyToSFTTableSafely(nameSFT){
+      setPTforSFT(nameSFT, Some(rcus), None)
+    }
+  }
+
+  def setWritesSFT(nameSFT: String, wcus: Long): Unit = {
+    applyToSFTTableSafely(nameSFT){
+      setPTforSFT(nameSFT, None, Some(wcus))
+    }
+  }
+
+  def setProvisionedThroughputCatalog(pt: ProvisionedThroughput): Unit = {
+    catalogTable.updateTable(pt)
+  }
+
+  def setProvisionedThroughputCatalog(rcus: Long, wcus: Long): Unit = {
+    setProvisionedThroughputCatalog(new ProvisionedThroughput(rcus, wcus))
+  }
+
+  def setReadsCatalog(rcus: Long): Unit = {
+    val currentPT = catalogTable.getDescription.getProvisionedThroughput
+    setProvisionedThroughputCatalog(rcus, currentPT.getWriteCapacityUnits)
+  }
+
+  def setWritesCatalog(wcus: Long): Unit = {
+    val currentPT = catalogTable.getDescription.getProvisionedThroughput
+    setProvisionedThroughputCatalog(currentPT.getReadCapacityUnits, wcus)
+  }
+
+  def getProvisionedThroughputCatalog: ProvisionedThroughputDescription = {
+    getDescriptionCatalog.getProvisionedThroughput
+  }
+
+  def getDescriptionCatalog: TableDescription = {
+    catalogTable.describe()
+  }
+
+}
+
+object DynamoDBDataStore {
+  val namespaceGlobal = "global"
+
+  val RCU_Key = "geomesa.dynamodb.sft.rcu"
+  val WCU_Key = "geomesa.dynamodb.sft.wcu"
+
+  val serId = "ser"
+
+  val geomesaKeyHash = "dtgandz2"
+  val geomesaKeyRange = "z3andID"
+
+  val featureKeySchema = List(
+    new KeySchemaElement().withAttributeName(geomesaKeyHash).withKeyType(KeyType.HASH),
+    new KeySchemaElement().withAttributeName(geomesaKeyRange).withKeyType(KeyType.RANGE)
+  )
+
+  val featureAttributeDescriptions = List(
+    new AttributeDefinition(geomesaKeyHash, ScalarAttributeType.B),
+    new AttributeDefinition(geomesaKeyRange, ScalarAttributeType.B)
+  )
+
+  val catalogKeyHash = "feature"
+  val catalogSftAttributeName = "sft"
+  val catalogNameSpaceAttributeName = "ns"
+
+  val catalogKeySchema = List(new KeySchemaElement(catalogKeyHash, KeyType.HASH))
+  val catalogAttributeDescriptions = List(new AttributeDefinition(catalogKeyHash, ScalarAttributeType.S))
+
+  def makeSFTTableName(catalog: String, nameSFT: String): String = {
+    val (ns, name) = SimpleFeatureTypes.buildTypeName(nameSFT)
+    val namespace = Option(ns).getOrElse(namespaceGlobal)
+    s"${catalog}_${namespace}_${name}_z3"
+  }
+
+  def getSchema(entry: ContentEntry, catalogTable: Table): SimpleFeatureType = {
+    val item = catalogTable.getItem("feature", entry.getTypeName)
```

Comment:
unit tests pass, but I feel that this shouldn't work given that I believe the namespace is not being used?